video/out/gpu/context.c

```diff
@@ -182,6 +186,10 @@ struct ra_ctx *ra_ctx_create(struct vo *vo, const char *context_type,
         };
 
         MP_VERBOSE(ctx, "Initializing GPU context '%s'\n", ctx->fns->name);
+        if (strcmp(ctx->fns->name, "wayland") == 0 && gnome) {
+            MP_WARN(ctx, "GNOME's wayland compositor is known to have many serious issues with mpv. Switch to GNOME's xorg session for the best experience.\n");
+            continue;
+        }
```

Comment:
This only detects opengl on wayland which is named `"wayland". Vulkan wayland is `"waylandvk"`. Also, I think you are leaking the ctx, given the `talloc_free(ctx)` that happens below for a non-match.

Reply:
Oh right, if you use `--gpu-api=vulkan`, it'll skip `wayland`.